akka-http/src/main/scala/akka/http/scaladsl/server/ContentNegotation.scala

```diff
@@ -22,9 +22,9 @@ final class MediaTypeNegotiator(requestHeaders: Seq[HttpHeader]) {
       Accept(mediaRanges) ← requestHeaders
       range ← mediaRanges
     } yield range).sortBy { // `sortBy` is stable, i.e. upholds the original order on identical keys
```

Comment:
Actually mulling this over a bit I think we should still take the `qValue` into account here, so the comments are actually correct and my change was wrong :).

This is a bit of a cornercase of a cornercase, but getting this right makes the solution easier to understand in the end IMO :). Added a testcase and fix in 7b3c4a4d22ad4e310805f74caf62b1d4e62f5624 but could use another pair of eyes :).